services/alerts/src/makeAlertsManager.ts

```diff
@@ -1,36 +1,96 @@
-import { Alert, AlertsManager, AlertsManagerAlert } from './types'
+import {
+    AlertsManager,
+    AlertsManagerAlert,
+    CreateAlertManagerAlertOptions,
+} from './types'
 
-type SetAlertsFunction = React.Dispatch<
-    React.SetStateAction<AlertsManagerAlert[]>
->
-
-const createAlertManagerAlert = (
-    alert: Alert,
-    id: number,
-    remove: (id: number) => void
-): AlertsManagerAlert => ({
-    ...alert,
+const createAlertManagerAlert = ({
+    id,
+    show,
+    hide,
+    remove,
+}: CreateAlertManagerAlertOptions): AlertsManagerAlert => ({
     id,
+    displayIndex: null,
+    hidden: true,
+    show,
+    hide: () => hide(id),
     remove: () => remove(id),
+    message: '',
+    options: {
+        hidden: true,
+    },
 })
 
+type AlertsManagerAlertsMap = Map<number, AlertsManagerAlert>
+
+const toVisibleAlertsArray = (alertsMap: AlertsManagerAlertsMap) =>
+    Array.from(alertsMap)
+        .reduce((alertsArray: Array<AlertsManagerAlert>, [_key, alert]) => {
+            if (alert.displayIndex) {
+                alertsArray.push(alert)
+            }
+            return alertsArray
+        }, [])
+        .sort(
+            (a: AlertsManagerAlert, b: AlertsManagerAlert) =>
+                (a.displayIndex || 0) - (b.displayIndex || 0)
+        )
+
+type SetAlertsFunction = React.Dispatch<
+    React.SetStateAction<AlertsManagerAlert[]>
+>
 export const makeAlertsManager = (
     setAlerts: SetAlertsFunction
 ): AlertsManager => {
+    const alertsMap: AlertsManagerAlertsMap = new Map()
     let id = 0
+    let displayIndex = 0
+
+    const show = (alert: AlertsManagerAlert) => {
+        displayIndex++
```

Comment:
I think this will increment the displayIndex with every call to `show`, even if the alert is already showing - is that the intended behavior?  I think this would cause the alerts to *jump* up to the top of the stack when `show` is called to update the message